### PR TITLE
Add user entity and roles enum, removed test stuff

### PR DESCRIPTION
**Pull request by Erikyy**

### CODE REVIEW

**Review by Orzelius — APPROVED** [review]
Looks good. I presume the CRUD stuff will come in a separate pr?

**Comment by Orzelius** [comment]
Ah it was WIP, sorry didn't notice as you had tagged me for review.

File: be/src/user/dto/user.dto.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import { IsNotEmpty, IsEmail } from 'class-validator';
+import { Role } from '../roles/user.roles';
+
+export class UserDTO {
```
Comment (Orzelius) [inline be/src/user/dto/user.dto.ts:6]: We don't need a UserDto, just use the original entity and add fields to it

File: be/src/user/dto/user.dto.ts [hunk 2]
```diff
@@ -0,0 +1,17 @@
+import { IsNotEmpty, IsEmail } from 'class-validator';
```
Comment (Orzelius) [inline be/src/user/dto/user.dto.ts:2]: I think class validator doesn't help here as gql will do the checking for us. Class-validator is used for the REST API

File: be/src/user/resolvers/user.resolver.ts [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    return await this.userService.findAll();
+  }
+
+  @Mutation(() => User)
+  async deleteUser(@Args('id') id: number) {
+    return await this.userService.delete(id);
+  }
+
+  @Mutation(() => User)
+  async createUser(
```
Comment (Orzelius) [inline be/src/user/resolvers/user.resolver.ts:47]: Use a createDTO/CreateInput here (example https://github.com/TPT-Loane/TPT-Loane/pull/82/commits/0613ae55228a11872744277c1dbda3d4d6489c90#diff-f91a82b2ed90eba19173f9242df0a6776b4dac8d4e75865fe443722e20430f85R13)

**Review by Orzelius — CHANGES_REQUESTED** [review]
This was a bit of a pain to review. Please follow the next principles in your next pr.

1. Small commits
2. Never change the code that was already committed (you commited a bunch of auto-gen you'll be changing in separate commits)
3. Commit auto-gen separately so it can be easily ignored

File: be/src/user/dto/user.dto.ts (hunk 1 above)
Comment (Erikyy) [reply be/src/user/dto/user.dto.ts:6]: NO NO NO, This will pose security risks, password hashes can be obtained from queries then.

Comment (Orzelius) [reply be/src/user/dto/user.dto.ts:6]: Just don't add a field to the pw? Also create a test that checks that the pw is not exposed, then nobody will add it by accident as well.

Comment (Erikyy) [reply be/src/user/dto/user.dto.ts]: Forgot to add test, adding that in a separate commit

**Review by Orzelius — APPROVED** [review]

**Pull request merged**